docs/configuration-files.rst

```diff
@@ -54,71 +54,86 @@ It defines and documents all supported sections and options. The file is located
 .. include:: ../data/anaconda.conf
     :code: ini
 
-Product configuration files
+Profile configuration files
 ---------------------------
 
-The product configuration files allow to override some of the configuration options for
-specific products. The files are located at ``/etc/anaconda/product.d/``.
+The profile configuration files allow to override some of the configuration options for
+specific profiles and products. The files are located at ``/etc/anaconda/profile.d/``.
 
 .. note::
 
     Anaconda previously used so called install classes for the product-specific configuration.
-    Install classes were completely removed and replaced by the product configuration files.
+    Install classes were completely removed and replaced by the profile configuration files.
+    These configuration files used to be called product configuration files for some time.
 
-Product identification
+Profile identification
 ^^^^^^^^^^^^^^^^^^^^^^
 
-The product is defined by a product name and optionally a variant name. For example, Fedora
-Server has a product name ``Fedora`` and a variant name ``Server``. See the terminology of
-`productmd <https://productmd.readthedocs.io/en/latest/terminology.html>`_.
+Each profile has a unique profile id. It is a lower-case string with no spaces that identifies
+the profile. The id can be arbitrary, but the convention is to use the name of the configuration
+file (for example, ``fedora-server``).
 
-The product can be specified by the boot options ``inst.product`` and ``inst.variant``.
-Otherwise, it will be automatically loaded from the ``.buildstamp`` file on the installation
-media. This file is generated by ``lorax`` using the ``--product`` and ``--variant`` options.
+Profile detection
+^^^^^^^^^^^^^^^^^
 
-Based on the provided product and variant names, the installer will look up the right
-configuration file in the ``/etc/anaconda/product.d/`` directory.
+The profile can be specified by the ``inst.profile`` boot option or the ``--profile`` cmdline
+option. Based on the provided profile id, the installer will look up the right configuration
+file in the ``/etc/anaconda/profile.d/`` directory.
```

Comment:
Shouldn't these be installed into `/usr/share/anaconda/profile.d` instead?

Reply:
Could you please specify reasoning why `/usr/share` instead of `/etc/`? I wonder what would be the benefit.

Reply:
Basically, if these are intended to be shipped as static files that users don't edit, they shouldn't be in `/etc`.

Reply:
Hmm, good point. However, to have that we would have to enable user to overload the configuration files from `/etc` especially now when there is Fedora Silverblue. We have something like that in `/etc/anaconda/conf.d/*.conf`. So the correct thing to do here is IMHO to move existing profiles to `/usr/share/...` and leave the `/etc/anaconda` support for overloading the conf files. And even in that case I'm not sure if user could overload the whole profile. I fear it would be a bit more work which shouldn't be part of this PR.

What do you think @poncovka ?

Reply:
I don't want to support overloading. Why users shouldn't edit existing profiles?

Reply:
I agree that this is work without a big benefit.
To @poncovka the main reason is that next RPM updated config wouldn't be applied because users has changed the config manually. However, in the Anaconda situation it doesn't have a big issue because there aren't many use-cases on the installed machine.

Reply:
I didn't expect overloading to be a thing, I just expect these to not be admin-editable generally.

Reply:
All our configuration files and drop dirs for configuration files are in `/etc/anaconda/`. Would you expect all of them to be in `/usr/share/`?

Reply:
Definitely not the `/etc/anaconda/conf.d/` users shouldn't place/edit anything in the `/usr/...` .

Reply:
In that case, I don't think that `conf.d` and `profile.d` should be at different locations. We support custom configurations for both of them, it is documented and encouraged. If we don't want to move all of these to `/usr/share/anaconda`, it might be better to keep using `/etc/anaconda/`.